packages/validator/src/util/polynomialCommitments.ts

```diff
@@ -0,0 +1,148 @@
+import {IChainForkConfig} from "@lodestar/config";
+import {blindedOrFullBlockHashTreeRoot} from "@lodestar/state-transition";
+import {allForks} from "@lodestar/types";
+import * as types from "@lodestar/types/eip4844";
+import {
+  BlobsSidecar,
+  KZGCommitment,
+  KZGProof,
+  PolynomialAndCommitment,
+  Polynomial,
+  BLSFieldElement,
+} from "@lodestar/types/lib/eip4844/sszTypes";
+
+// function bytesToNumber(bytes: Uint8Array): number {
+//   return Buffer.from(bytes).readUIntBE(0, bytes.length);
+// }
+
+// https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/validator.md#hash_to_bls_field
+// def hash_to_bls_field(x: Container) -> BLSFieldElement:
+//     """
+//     Compute 32-byte hash of serialized container and convert it to BLS field.
+//     The output is not uniform over the BLS field.
+//     """
+//     return bytes_to_bls_field(hash(ssz_serialize(x)))
+export function hashToBlsField(_x: types.PolynomialAndCommitment): types.BLSFieldElement {
+  return BLSFieldElement.defaultValue();
+}
+
+// https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/polynomial-commitments.md#blob_to_kzg_commitment
+// def blob_to_kzg_commitment(blob: Blob) -> KZGCommitment:
+//     return g1_lincomb(bit_reversal_permutation(KZG_SETUP_LAGRANGE), blob)
+export function blobToKzgCommitment(_blob: types.Blob): types.KZGCommitment {
+  return KZGCommitment.defaultValue();
+}
+
+// https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/polynomial-commitments.md#evaluate_polynomial_in_evaluation_form
+// def evaluate_polynomial_in_evaluation_form(polynomial: Sequence[BLSFieldElement],
+//                                            z: BLSFieldElement) -> BLSFieldElement:
+//     """
+//     Evaluate a polynomial (in evaluation form) at an arbitrary point `z`
+//     Uses the barycentric formula:
+//        f(z) = (1 - z**WIDTH) / WIDTH  *  sum_(i=0)^WIDTH  (f(DOMAIN[i]) * DOMAIN[i]) / (z - DOMAIN[i])
+//     """
+//     width = len(polynomial)
+//     assert width == FIELD_ELEMENTS_PER_BLOB
+//     inverse_width = bls_modular_inverse(width)
+
+//     # Make sure we won't divide by zero during division
+//     assert z not in ROOTS_OF_UNITY
+
+//     roots_of_unity_brp = bit_reversal_permutation(ROOTS_OF_UNITY)
+
+//     result = 0
+//     for i in range(width):
+//         result += div(int(polynomial[i]) * int(roots_of_unity_brp[i]), (z - roots_of_unity_brp[i]))
+//     result = result * (pow(z, width, BLS_MODULUS) - 1) * inverse_width % BLS_MODULUS
+//     return result
+export function evaluatePolynomialInEvaluationForm(
+  _polynomial: types.Polynomial,
+  _z: types.BLSFieldElement
+): types.BLSFieldElement {
+  // This is now in https://pkg.go.dev/github.com/protolambda/go-kzg@v0.0.0-20221025081131-f3a74d3b1d0c/bls?utm_source=gopls#EvaluatePolyInEvaluationForm
+  // We will use a KZG library for this!
```

Comment:
It's now implemented in Proto's go-kzg https://github.com/protolambda/go-kzg/blob/master/bls/globals.go#L105.

I'm not aware of an implementation anywhere else.

First, need to figure out NodeJS bindings for c-kzg... Then we can port the Golang implementation there?